File: 0125-valid-palindrome/0125-valid-palindrome.scala
object Solution {
    def isPalindrome(s: String): Boolean = {
        var start = 0
        var end = s.length - 1

        while(start <= end){
            val i = s.charAt(start)
            val j = s.charAt(end)

            if(!Character.isLetterOrDigit(i)){
                start +=1
            }
            else if(!Character.isLetterOrDigit(j)){
                end -=1
            } 
            else{
                if(Character.toLowerCase(i) != Character.toLowerCase(j)){
                    return false
                }
                else{
                    start +=1
                    end -=1
                }
            }
        }
        true
    }
}

File: 0125-valid-palindrome/0125-valid-palindrome.java
class Solution {
    public boolean isPalindrome(String s) {
        int start = 0;
        int end = s.length() -1;
        
        while(start<=end){
            char I = s.charAt(start);
            char J = s.charAt(end);
            if(!Character.isLetterOrDigit(I)){
                start++;
            }
            else if(!Character.isLetterOrDigit(J)){
                end--;
            }
            else{
                if(Character.toLowerCase(I)!= Character.toLowerCase(J)){
                    return false;
                }
                else{
                    start++;
                    end --;
                }
            }
        }
        return true;
    }
}